app/components/RuleViewerEditor/subcomponents/RuleInputOutputFieldsComponent.tsx

```diff
@@ -82,17 +85,27 @@ export default function RuleInputOutputFieldsComponent({
       if (!draft.content?.fields) {
         draft.content = { fields: [] };
       }
-      draft.content.fields.push({ id: crypto.randomUUID(), field: "" });
+      draft.content.fields.push({ name: "" });
       return draft;
     });
   };
 
-  const updateInputField = (item: InputOutputField, { key, label }: any) => {
+  const updateInputField = async (item: InputOutputField, { value, label }: any) => {
+    // Get the actual values from the ids
+    // TODO: potentially add loading feedback when this is being fetched
+    const data: KlammBREField = await getBREFieldFromName(value);
+    // Update the node with the information we want to store in the json
     updateNode(id, (draft) => {
       draft.content.fields = draft.content.fields.map((input: InputOutputField) => {
+        console.log(input.id, item.id, data.id);
         if (input.id === item.id) {
-          input.field = key;
-          input.label = label;
+          // Get important bits of data to store in json
+          input.id = data.id;
+          input.name = data.name;
+          input.label = data.label;
+          input.description = data.description;
+          input.dataType = data?.data_type?.name;
+          input.validationCriteria = data?.data_validation?.validation_criteria;
```

Comment:
This looks awesome and provides so many useful instructions for us to use! Would we also be able to add the data?.data_validation?.bre_validation_type?.value, as this would be the validation instructions, such as '>='.

Reply:
Good catch! Updated.